books/RayTracingInOneWeekend.html

```diff
@@ -502,54 +502,103 @@
+  (this is also commonly referred to as the _eye point_).
+The vector from the camera center to the viewport center will be orthogonal to the viewport.
+We'll initially set the distance between the viewport and the camera center point to be one unit.
+This distance is often referred to as the _focal length_.
+
+I’ll put the camera center at $(0,0,0)$.
 I will have the y-axis go up, and the x-axis to the right.
-In order to respect the convention of a right handed coordinate system, into the screen is the
-negative z-axis.
+Since we use a right-handed coordinate system, the vector into the screen is the negative z-axis.
+Note that I do not make the ray direction a unit length vector because I think not doing that makes
+  for simpler and slightly faster code.
+
+  ![Figure [camera-geom]: Camera geometry](../images/fig-1.03-cam-geom.jpg)
 
 Now the inevitable tricky part.
 While our 3D space has the conventions above, this conflicts with the standard _image_ coordinates,
-where the Y axis is inverted.
-In image coordinates, Y=0 denotes the top of the image, and Y=1 denotes the bottom.
-
-To help keep things straight, we'll compute two normalized (that is, in the range [0,1]) image
-coordinates: $s$ (horizontal) and $t$ (vertical).
-$s=0$ is image left, $s=1$ is image right, $t=0$ is image top, and $t=1$ is image bottom.
-That means that we get our vertical axis flip at the right point by computing $t = 1 - y$, where $y$
-is the vertical component of our 3D coordinate system, and $t$ is the flipped vertical component of
-our image coordinate system.
-
-I will traverse the screen from the upper left hand corner, and use two offset vectors along the
-screen sides to move the ray endpoint across the screen.
-Note that I do not make the ray direction a unit length vector because I think not doing that makes
-for simpler and slightly faster code.
+  where the Y axis is inverted (increasing going down).
+
+As we scan our image, we will start at the upper left pixel (pixel 0,0), scan left-to-right across
+  the row, and then scan the next row, top-to-bottom (reading order for most Latin languages).
+To help navigate the pixel grid, we'll use a vector from the left edge to the right edge
+  (`viewport_u`), and a vector from the upper edge to the lower edge (`viewport_v`).
 
-  ![Figure [cam-geom]: Camera geometry](../images/fig-1.03-cam-geom.jpg)
+Our pixel grid will be inset from the viewport edges by half the pixel-to-pixel distance, so each
+  pixel will gather light from the area around it.
+Here's what our viewport and pixel grid look like:
+
+  ![Figure [pixel-grid]: Viewport and pixel grid](../images/fig-1.04-pixel-grid.jpg)
+
+In this figure, we have the viewport, the pixel grid for a 7×5 resolution image, the viewport upper
+  left corner $\mathbf{Q}$, the pixel $\mathbf{P_{0,0}}$ location, the `viewport_u` vector
+  $\mathbf{V_u}$, the `viewport_v` vector $\mathbf{V_v}$, and the pixel delta vectors
+  $\mathbf{\Delta_u}$ and $\mathbf{\Delta_v}$.
```

Comment:
$\mathbf{\Delta u}$ and $\mathbf{\Delta v}$.

Reply:
Done.